hostplumber/main.go

```diff
@@ -51,6 +53,35 @@ func init() {
 	//+kubebuilder:scaffold:scheme
 }
 
+func validateIPAddressPort(address string) error {
+
+	host, portStr, err := net.SplitHostPort(address)
+	if err != nil {
+		return fmt.Errorf("address must be in the format IP:port")
+	}
+
+	// Validate the IP part
+	parsedIP := net.ParseIP(host)
+	if parsedIP == nil {
+		return fmt.Errorf("invalid IP address")
+	}
+
+	if parsedIP.To4() == nil && parsedIP.To16() == nil {
+		return fmt.Errorf("metric bind address isn't ipv4 or ipv6 address")
+	}
+
+	// Validate the port part
+	port, err := strconv.Atoi(portStr)
```

Comment:
I don't think you need these checks. You probably need the check in line 64, since `SplitHostPort` also accepts a hostname as a valid object, but the other checks should be taken care off by the function itself.

From its documentation:
> See func Dial for a description of the hostport parameter, and host and port results.

The [Dial](https://pkg.go.dev/net#Dial) function does this:
> For TCP and UDP networks, the address has the form "host:port". The host must be a literal IP address, or a host name that can be resolved to IP addresses. The port must be a literal port number or a service name. If the host is a literal IPv6 address it must be enclosed in square brackets, as in "[2001:db8::1]:80" or "[fe80::1%zone]:80". The zone specifies the scope of the literal IPv6 address as defined in [RFC 4007](https://rfc-editor.org/rfc/rfc4007.html). The functions [JoinHostPort](https://pkg.go.dev/net#JoinHostPort) and [SplitHostPort](https://pkg.go.dev/net#SplitHostPort) manipulate a pair of host and port in this form. When using TCP, and the host resolves to multiple IP addresses, Dial will try each IP address in order until one succeeds.

Reply:
Removing the validation as per discussion with @mithilarun